src/orion/algo/hyperband.py

```diff
@@ -279,12 +330,14 @@ def register_samples(self, bracket, samples):
                 self.get_id(sample, ignore_fidelity=True, ignore_parent=True)
                 not in self.trial_to_brackets
             ):
+                assert self.brackets is not None
                 self.trial_to_brackets[
                     self.get_id(sample, ignore_fidelity=True, ignore_parent=True)
                 ] = self.brackets.index(bracket)
 
-    def promote(self, num):
-        samples = []
+    def promote(self, num: int) -> list[Trial]:
+        samples: list[Trial] = []
+        assert self.brackets is not None
```

Comment:
Same here

Reply:
Fixed in 35ce6867769d77028c87f7bb015adfd349315f85